core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -144,6 +145,9 @@ protected void doExecute(Task task, BulkRequest bulkRequest, ActionListener<Bulk
             // Attempt to create all the indices that we're going to need during the bulk before we start.
             // Step 1: collect all the indices in the request
             final Set<String> indices = bulkRequest.requests.stream()
+                .filter(request -> request.opType() != DocWriteRequest.OpType.DELETE 
+                        || request.versionType() == VersionType.EXTERNAL 
```

Comment:
can you add a comment here as to why we have a special handling for external versions and deletes here?